backend/clubs/serializers.py

```diff
@@ -1249,8 +1249,16 @@ def get_approved_by(self, obj):
         return obj.approved_by.get_full_name()
 
     def get_advisor_set(self, obj):
+        user = self.context["request"].user
+
+        public_filter = (
+            obj.advisor_set.filter(public__gte=Advisor.PUBLIC_STUDENTS).order_by("name")
+            if user.is_authenticated
+            else obj.advisor_set.filter(public=Advisor.PUBLIC_ALL).order_by("name")
+        )
```

Comment:
nit: could do something like
```suggestion
        visibility_level = Advisor.PUBLIC_STUDENTS if user.is_authenticated else Advisor.PUBLIC_ALL
        queryset = obj.advisor_set.filter(public__gte=visibility_level)
```

Reply:
Great catch, much cleaner!